runtime/gc_base/IndexableObjectAllocationModel.cpp

```diff
@@ -145,7 +145,7 @@ MM_IndexableObjectAllocationModel::initializeIndexableObject(MM_EnvironmentBase
 	/* Lay out arraylet and arrayoid pointers */
 	switch (_layout) {
 	case GC_ArrayletObjectModel::InlineContiguous:
-		Assert_MM_true(1 == _numberOfArraylets);
+		Assert_MM_true((_dataSize == 0) || (1 == _numberOfArraylets));
```

Comment:
We need comment before this assertion explained (_dataSize == 0) for InlineContiguous is possible for 0-stride elements of Flattened Array

Reply:
also please put constant first (0 == _dataSize)